include/linux/kfence.h

```diff
@@ -20,8 +34,12 @@ bool kfence_discard_slab(struct kmem_cache *s, struct page *page);
 
 bool kfence_handle_page_fault(unsigned long addr);
 
-// TODO(elver): Make is_kfence_addr inlinable.
-bool is_kfence_addr(void *addr);
+static inline bool is_kfence_addr(void *addr)
+{
+	const unsigned long uaddr = (unsigned long)addr;
+
+	return unlikely(uaddr >= __kfence_pool_start && uaddr < __kfence_pool_end());
```

Comment:
One potential optimization is to allocate pages at fixed virtual address and have a static max bound of number of pages, then this check can be expressed as: `(uaddr - KFENCE_POOL_START) < KFENCE_MAX_POOL_SIZE`. This is one branch and does not involve any memory accesses.